spec/compiler/crystal/tools/tracing_spec.cr

```diff
@@ -0,0 +1,104 @@
+require "spec"
+require "compiler/crystal/tools/tracing"
+require "../../../support/tempfile"
+
+describe Crystal::Tracing::StatsCommand do
+  it "parses stdin" do
+    stdin = IO::Memory.new <<-TRACE
+    gc malloc t=410557 d=9012 size=16
+    gc malloc t=482582 d=401 size=4 atomic=1
+    gc malloc t=493775 d=128 thread=7f0ea62bd740 size=144
+    sched spawn t=520355 d=-1 thread=7f0ea62bd740 fiber=7f0ea6299f00
+    sched enqueue t=530598 d=101572 thread=7f0ea62bd740 fiber=7f0ea6299e60 [main] fiber=7f0ea6299f00
+    gc malloc t=550326 d=222 thread=7f0ea62bd740 size=24
+    gc collect t=561057 d=79489 thread=7f0ea62bd740
+    gc heap_resize t=102125791993927 d=-1 thread=7f0ea62bd740 size=131072
+    TRACE
+    stdout = IO::Memory.new
+    stderr = IO::Memory.new
+
+    Crystal::Tracing::StatsCommand.new("-", stdin: stdin, stdout: stdout, stderr: stderr).run
+    output = stdout.to_s
+    output.should contain("gc:malloc events=4 durations=")
+    output.should contain("gc:collect events=1 durations=")
+    output.should contain("gc:heap_resize events=1")
+    output.should contain("sched:spawn events=1")
+    output.should contain("sched:enqueue events=1 durations=")
+    stderr.to_s.should be_empty
+  end
+
+  it "--fast" do
+    stdin = IO::Memory.new <<-TRACE
+    gc malloc t=410557 d=9012 size=16
+    gc malloc t=482582 d=401 size=4 atomic=1
+    gc malloc t=493775 d=128 thread=7f0ea62bd740 size=144
+    sched spawn t=520355 d=-1 thread=7f0ea62bd740 fiber=7f0ea6299f00
+    sched enqueue t=530598 d=101572 thread=7f0ea62bd740 fiber=7f0ea6299e60 [main] fiber=7f0ea6299f00
+    gc malloc t=550326 d=222 thread=7f0ea62bd740 size=24
+    gc collect t=561057 d=79489 thread=7f0ea62bd740
+    gc heap_resize t=102125791993927 d=-1 thread=7f0ea62bd740 size=131072
+    TRACE
+    stdout = IO::Memory.new
+    stderr = IO::Memory.new
+
+    Crystal::Tracing::StatsCommand.new("-", fast: true, stdin: stdin, stdout: stdout, stderr: stderr).run
+    output = stdout.to_s
+    output.should contain("gc:malloc events=4 duration=")
+    output.should contain("gc:collect events=1 duration=")
+    output.should contain("gc:heap_resize events=1")
+    output.should contain("sched:spawn events=1")
+    output.should contain("sched:enqueue events=1 duration=")
```

Comment:
this is the same as without --fast. what does change?

Reply:
Oh, both specs need to check the stats :facepalm: